src/java.base/share/classes/java/lang/invoke/StringConcatFactory.java

```diff
@@ -998,4 +1032,293 @@ private static MethodHandle stringValueOf(Class<?> ptype) {
     private StringConcatFactory() {
         // no instantiation
     }
+
+    /**
+     * Simplified concatenation method to facilitate {@link StringTemplate}
+     * concatenation. This method returns a single concatenation method that
+     * interleaves fragments and values. fragment|value|fragment|value|...|value|fragment.
+     * The number of fragments must be one more that the number of ptypes.
+     * The total number of slots used by the ptypes must be less than or equal
+     * to MAX_INDY_CONCAT_ARG_SLOTS.
```

Comment:
see my comment about making MAX_INDY_CONCAT_ARG_SLOTS public

Reply:
Disagree.

Reply:
As i said above, i consider this thread as resolved

Reply:
Changing